### PR TITLE
[dotnet] - remove ubuntu:noble's ubuntu user capturing 1000 uid & gid

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Devcontainer Names**

* dotnet

**Description**

* This PR attemtps to make the following changes:
	- Removes the existing `ubuntu` user provided by the base-image for `ubuntu-noble` base image i.e. existing user blocking uid & gid 1000
	
**_Changelog_** 

* Changed the `Dockerfile` and wrote a test case for the devcontainer image if ubuntu:noble is used as base image o.s

**Checklist**

* [x] Checked that applied changes work as expected

### CODE REVIEW

File: src/dotnet/test-project/test.sh [hunk 1]
```diff
@@ -29,5 +29,23 @@ check "usr-local-etc-config-does-not-exist" test ! -f "/usr/local/etc/gitconfig"
 
 checkPackageVersion "pwsh" "7.4.4" "PowerShell"
 
+check_ubuntu_user() {
```
Comment (samruddhikhandale) [inline src/dotnet/test-project/test.sh:32]: Thanks for adding these, however, let's remove these tests as we would never update the default image to be `noble`

File: src/dotnet/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -2,6 +2,15 @@ ARG VARIANT=9.0-bookworm-slim
 FROM mcr.microsoft.com/dotnet/sdk:${VARIANT}
 ENV PATH $PATH:/home/vscode/.dotnet:/home/vscode/.dotnet/tools
 
+ARG VARIANT
+RUN if [ "${VARIANT#*noble}" != "$VARIANT"  ]; then \
+        if id "ubuntu" &>/dev/null; then \
```
Comment (samruddhikhandale) [inline src/dotnet/.devcontainer/Dockerfile:7]: Testing these changes and pushing to dev tags
https://github.com/devcontainers/images/actions/runs/11020617403

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/dotnet/.devcontainer/Dockerfile [hunk 3]
```diff
@@ -2,6 +2,15 @@ ARG VARIANT=9.0-bookworm-slim
 FROM mcr.microsoft.com/dotnet/sdk:${VARIANT}
 ENV PATH $PATH:/home/vscode/.dotnet:/home/vscode/.dotnet/tools
 
+ARG VARIANT
+RUN if [ "${VARIANT#*noble}" != "$VARIANT"  ]; then \
```
Comment (samruddhikhandale) [inline src/dotnet/.devcontainer/Dockerfile:6]: Can you add back these changes which pins to 1000? 

https://github.com/devcontainers/images/pull/1195/files#diff-cc0cf9f2ed094e192be020fc2105581e6a56adde975fe41251b6c7debc8f6cf4L10-L12

Comment (gauravsaini04) [reply src/dotnet/.devcontainer/Dockerfile:6]: sure

File: src/dotnet/test-project/test.sh (hunk 1 above)
Comment (gauravsaini04) [reply src/dotnet/test-project/test.sh:32]: ok

**Review by samruddhikhandale — APPROVED** [review]

**Pull request merged**